src/websiteAccount/WebsiteAccount.java

```diff
@@ -124,46 +132,68 @@ public static boolean addWebsiteAccountPrompts(ConnectionSource databaseConnecti
 		return websiteAccount.addWebsiteAccount(databaseConnection);
 	}
 	
-	public static String viewWebsiteAccountLogin(ConnectionSource databaseConnection, Scanner keyboard, User safeStoreUser) {
+	/**
+	 * gets the foreign collection of all website account rows for that user
+	 * @param databaseConnection
+	 * @param safeStoreUser of type User
+	 * @return a ForeignCollection<WebsiteAccountEntity> that holds all the db rows of accounts
+	 * 			for the safeStoreUser
+	 */
+	public static ForeignCollection<WebsiteAccountEntity> getAllWebsiteAccounts(ConnectionSource databaseConnection, User safeStoreUser) {
+		Dao<UserEntity, String> userDao;
 		try {
-			// get user object from db to view accounts
-			Dao<UserEntity, String> userDao = DaoManager.createDao(databaseConnection, UserEntity.class);
-			UserEntity returnedFromDb = userDao.queryForSameId(safeStoreUser.getUserEntity());
-			ForeignCollection<WebsiteAccountEntity> websiteAccounts = returnedFromDb.getWebsiteAccounts();
-			
-			// print out all the accounts for the user
-			if (websiteAccounts.isEmpty()) {
-				System.out.println("No accounts");
-				return "No accounts";
-			}
+			userDao = DaoManager.createDao(databaseConnection, UserEntity.class);
+			UserEntity userEntity = userDao.queryForSameId(safeStoreUser.getUserEntity());
+			return userEntity.getWebsiteAccounts();
+		} catch (SQLException e) {
+			e.printStackTrace();
+			return null;
+		}
+	}
+	
+	/**
+	 * Prints a list of all website accounts for the User
+	 * @param databaseConnection
+	 * @param safeStoreUser of type User
+	 */
+	public static void printAllWebsiteAccounts(ConnectionSource databaseConnection, User safeStoreUser) {
+		ForeignCollection<WebsiteAccountEntity> websiteAccounts = getAllWebsiteAccounts(databaseConnection, safeStoreUser);
+		if (websiteAccounts.isEmpty() || websiteAccounts == null) {
+			System.out.println("No accounts");
+		}
+		for (WebsiteAccountEntity account : websiteAccounts) {
+			System.out.println(Encryption.decrypt(account.getNickname()));
+		}
+	}
+	
+	/**
+	 * Prompts the user to pick the website account they want to view the login and password info for and prints it out
+	 * @param databaseConnection
+	 * @param keyboard
+	 * @param safeStoreUser
+	 * @return - what was printed for that specific website account...mainly used for testing
+	 */
+	public static String viewWebsiteAccountInfo(ConnectionSource databaseConnection, Scanner keyboard, User safeStoreUser) {
+		printAllWebsiteAccounts(databaseConnection, safeStoreUser);
+		ForeignCollection<WebsiteAccountEntity> websiteAccounts = getAllWebsiteAccounts(databaseConnection, safeStoreUser);
+
+		String returnAccountInfo = "No accounts";
+		Boolean accountExists = false;
+		System.out.println("Type the account that you want to see the login info for:");
+		while (!accountExists) {
```

Comment:
Should there be a way for this while loop to terminate if the user doesn't have any accounts saved (in which case the call to printAllWebsiteAccounts(databaseConnection, safeStoreUser would print No accounts but then there's no way for the user to get out of this while loop)

Reply:
Yes you are right....I added an if statement that if there are no accounts, the method will return early